src/App.js

```diff
@@ -136,29 +137,34 @@ class App extends React.PureComponent<Props, State> {
                 <CloseContext.Provider value={this.props.onClose}>
                   <UserContext.Provider value={this.state.userContext}>
                     <SearchStateProvider>
-                      <BookingStateProvider onLogout={this.props.onLogout}>
-                        <ExtraInfoStateProvider>
-                          <ThemeProvider>
-                            {isOpen &&
-                              route && (
-                                <EventListener
-                                  target="window"
-                                  onKeydown={withOptions(this.onKeyDown, {
-                                    capture: true,
-                                  })}
-                                >
-                                  <SelectUrlBooking
-                                    wasSelected={
-                                      this.state.urlBookingWasSelected
-                                    }
-                                    setSelected={this.urlBookingSelected}
-                                  />
-                                  <Routes route={route} />
-                                </EventListener>
-                              )}
-                          </ThemeProvider>
-                        </ExtraInfoStateProvider>
-                      </BookingStateProvider>
+                      <SelectedBookingProvider>
+                        <BookingStateProvider
+                          hasBooking={false}
```

Comment:
why `BookingStateProvider` needs property `hasBooking` if it's always false?

Reply:
It's not going to be always false. When the user is logged in and has a booking, it's going to be true. I'm just trying to make small steps because it's very easy to break stuff and get lost in debugging here